make/modules/jdk.jpackage/Lib.gmk

```diff
@@ -44,27 +58,59 @@ endif
 
 JPACKAGE_OUTPUT_DIR := $(JDK_OUTPUTDIR)/modules/$(MODULE)/jdk/jpackage/internal/resources
 JPACKAGE_CXXFLAGS_windows := -EHsc -DUNICODE -D_UNICODE
+JPACKAGE_CFLAGS_windows := -DUNICODE -D_UNICODE
+JPACKAGE_APPLAUNCHER_INCLUDES := $(addprefix -I, \
+    $(sort $(JPACKAGE_APPLAUNCHER_SRC) $(JPACKAGE_APPLAUNCHERLIB_SRC)) \
+)
 
 # Output app launcher executable in resources dir, and symbols in the object dir
 $(eval $(call SetupJdkExecutable, BUILD_JPACKAGE_APPLAUNCHEREXE, \
     NAME := jpackageapplauncher, \
     OUTPUT_DIR := $(JPACKAGE_OUTPUT_DIR), \
     SYMBOLS_DIR := $(SUPPORT_OUTPUTDIR)/native/$(MODULE)/jpackageapplauncher, \
     SRC := $(JPACKAGE_APPLAUNCHER_SRC), \
-    TOOLCHAIN := TOOLCHAIN_LINK_CXX, \
+    TOOLCHAIN := $(JPACKAGE_APPLAUNCHER_TOOLCHAIN), \
     OPTIMIZATION := LOW, \
     CXXFLAGS := $(call JpackageWithStaticCrt, $(CXXFLAGS_JDKEXE)) \
-        $(addprefix -I, $(JPACKAGE_APPLAUNCHER_SRC)), \
+        $(JPACKAGE_APPLAUNCHER_INCLUDES), \
+    CFLAGS := $(call JpackageWithStaticCrt, $(CFLAGS_JDKEXE)) \
+        $(JPACKAGE_APPLAUNCHER_INCLUDES), \
+    CFLAGS_linux := -Wno-format-nonliteral, \
+    CFLAGS_macosx := -Wno-format-nonliteral, \
     CXXFLAGS_windows := $(JPACKAGE_CXXFLAGS_windows), \
+    CFLAGS_windows := $(JPACKAGE_CFLAGS_windows), \
     LDFLAGS := $(LDFLAGS_JDKEXE), \
-    LIBS_macosx := -framework Cocoa, \
-    LIBS := $(LIBCXX), \
-    LIBS_linux := -ldl, \
+    LIBS_macosx := $(LIBCXX) -framework Cocoa, \
+    LIBS_windows := $(LIBCXX), \
+    LIBS_linux := -nodefaultlibs -lc -ldl, \
 ))
 
-$(BUILD_JPACKAGE_APPLAUNCHEREXE): $(call FindLib, java.base, java)
+JPACKAGE_TARGETS += $(BUILD_JPACKAGE_APPLAUNCHEREXE)
+
+
+################################################################################
+
+ifeq ($(call isTargetOs, linux), true)
+
+  $(eval $(call SetupJdkLibrary, BUILD_JPACKAGE_APPLAUNCHERLIB, \
+      NAME := jpackageapplauncherlib, \
+      OUTPUT_DIR := $(JPACKAGE_OUTPUT_DIR), \
+      SYMBOLS_DIR := $(SUPPORT_OUTPUTDIR)/native/$(MODULE)/jpackageapplauncherlib, \
+      SRC := $(JPACKAGE_APPLAUNCHERLIB_SRC), \
+      TOOLCHAIN := TOOLCHAIN_LINK_CXX, \
+      OPTIMIZATION := LOW, \
+      CXXFLAGS := $(CXXFLAGS_JDKLIB) $(JPACKAGE_APPLAUNCHER_INCLUDES), \
+      CFLAGS := $(CFLAGS_JDKLIB) $(JPACKAGE_APPLAUNCHER_INCLUDES), \
+      CFLAGS_linux := -Wno-format-nonliteral, \
+      LDFLAGS := $(LDFLAGS_JDKLIB) \
+          -Wl$(COMMA)--version-script=$(TOPDIR)/make/modules/$(MODULE)/applauncherlib.ld-version-script, \
```

Comment:
We should really not be using linker scripts. I did not understand your comment in the linker script -- was it only needed to handle your personal build environment? If so, you need to fix your build environment instead.

Reply:
Yeh, I found that linker script is needed only in my local build env. I'll remove it then.